x-pack/plugins/security_solution/docs/prebuilt_rules_customization_rfc.md

```diff
@@ -0,0 +1,2867 @@
+        <td><code>Prebuilt rule name</code></td>
+        <td><code>Customized prebuilt rule name</code></td>
+        <td style="border-right:3px solid black"><code>Updated prebuilt rule name. Much better.</code></td>
+        <td><code>Customized prebuilt rule name. Much better.</code></td>
+      </tr>
+    </tbody>
+  </table>
+
+
+##### Number fields
+
+Number fields should be treated as a whole unit, i.e. not breakable by digits. Therefore, there is only one possibility of conflicts, the scenario (A B C). In that case, **keep the current version** with the user customization.
+
+Examples: `risk_score`, `max_signals`
+
+  <table>
+    <thead>
+      <tr>
+        <th style="border-right:3px solid black">Use case</th>
+        <th>Base version</th>
+        <th>Current version</th>
+        <th style="border-right:3px solid black">Target version</th>
+        <th>Merged version (output)</th>
+      </tr>
+    </thead>
+    <tbody align="center">
+      <tr>
+        <td style="border-right:3px solid black">Keep customizations if conflict</td>
+        <td><code>50</code></td>
+        <td><code>65</code></td>
+        <td style="border-right:3px solid black"><code>20</code></td>
+        <td><code>65</code></td>
+      </tr>
+    </tbody>
+  </table>
+
+##### Array fields
+
+###### Array of strings fields
+
+For **array of strings** fields, conflict resolution will take place on an element by element basis - that means that the algorithm will not consider or try to solve changes within a single element. For example:
+
+```
+base: [test1]
+current: [my-test1]
+target: [test2]
+output: [my-test1]
+```
+Therefor the example above, the algorithm should not try to merge the single string element by breaking it into parts.
+
+Example **array of strings** fields: `index`, `tags`, `references`,
+
+###### Array of objects fields
+
+Since the properties in each object within the array are highly dependent on one another, the whole object should be treated as a block, and not attempt to diff or merge changes to individual fields within each element.
+
+For example: in a `required_fields` update, if the `type` is updated from `unknown` in the base version to `keyword` in the `target` version, but in the same object the user has updated the base `name` of the field, the change of type will have a high probability of not making sense anymore - there is no way of knowing if the change proposed by Elastic will also apply to the field name that the user is now referring through its customization.
+
+Therefore, we should prefer to keep the user's version for the whole block.(See this example in the first element of the array in the first row of the table below).
+
+Examples: `related_integrations,` `required_fields`
+
+###### Proposed algorithm
+
+- Do element by element comparison:
+  - If element index exists in the three versions and the values:
+    - do not differ between all versions (base, current and target), **keep current** version. (A A A)
```

Comment:
It could be an explanation or a link for another doc somewhere what designation `(A A A)` mean.